lib/trie.js

```diff
@@ -0,0 +1,57 @@
+module.exports = Trie;
+
+function Trie() {
+  this.methods = {}
+}
+
+//insert new nodes to Trie
+Trie.prototype.add = function (path, verb, handler) {
+  var node = this;
+  var parts = getPathParts(path);
+
+  if (parts.length) {
+    for (var i = 0; i < parts.length; i++) {
+      if (!node || !(parts[i] in node)) {
+        node[parts[i]] = new Trie();
+      }
+      if (i + 1 === parts.length) {
+        node[parts[i]].methods = node[parts[i]].methods || {};
+        node[parts[i]].methods[verb] = handler;
```

Comment:
L18-19 and L25-26 are duplicated, please extract a shared private method.

An example of a possible API in action:

```js
node[parts[i]].addVerbHandler(verb, handler);
```